druid/src/core.rs

```diff
@@ -268,10 +268,16 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
         paint_ctx.z_ops.append(&mut ctx.z_ops);
 
         if env.get(Env::DEBUG_PAINT) {
-            let rect = Rect::from_origin_size(Point::ORIGIN, ctx.size());
+            const BORDER_WIDTH: f64 = 1.0;
+            const BORDER_MIDPOINT: f64 = BORDER_WIDTH / 2.0;
+            let size = ctx.size();
+            let rect = Rect::from_origin_size(
+                (BORDER_MIDPOINT, BORDER_MIDPOINT),
+                (size.width - BORDER_WIDTH, size.height - BORDER_WIDTH),
+            );
```

Comment:
another way to think about this  is in terms of insets:

```rust
const BORDER_WIDTH: f64 = 1.0;
let bounds = ctx.size().to_rect();
let stroke_inset = Insets::uniform(BORDER_WIDTH / 2.0);
let stroke_rect = rect - stroke_inset;
```

Reply:
The inset helper methods are definitely helpful and made the code more concise. In this case the insets do have to be negative though, to shrink the rect.